### ISSUE TITLE
understanding where vcs out come from....

### ISSUE THREAD

**Issue by jD4rk** [romkatv/powerlevel10k#478]
I might found an behavior about displaying the right prompt inside **git directory** when there is merge conflict.

by setting in .p10.zsh the follow
  **typeset -g POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true**

the correct prompt in a merge conflict git directory isn't show.
I guess because of this: (_source p10k.zsh_)
> $_POWERLEVEL9K_VCS_CONFLICTED_STATE == 1

This test seems to be always false

>  ❯ typeset -m '_POWERLEVEL9K_VCS_CONFLICTED_*'
> _POWERLEVEL9K_VCS_CONFLICTED_STATE=0
> _POWERLEVEL9K_VCS_CONFLICTED_MAX_NUM=-1
> _POWERLEVEL9K_VCS_CONFLICTED_BACKGROUND=1

The two screenshots show that once you get rid of this bad condition the behavior works as aspected

<img width="1124" alt="ok" src="https://user-images.githubusercontent.com/8858312/74059841-0a770000-49e9-11ea-9deb-50659afea07b.png">
<img width="1130" alt="error" src="https://user-images.githubusercontent.com/8858312/74059849-0ea31d80-49e9-11ea-974d-45249f91bbc5.png">

PS:
in the PR I also change few icons (commit ahead/behind) because I think they matching more with the pattern of the other used for untracked/unstaged

**Comment by romkatv** [comment]
Please provide the following information:

1. Your complete powerlevel10k configuration that includes all `POWERLEVEL9K_*` parameters and any functions that may be referred to by these parameters. If your powerlevel10k is contained within `~/.p10k.zsh`, please attach that file.
2. A screenshot of your prompt.
3. An explanation of what you believe is incorrect in the screenshot.

**Comment by jD4rk** [comment]
Ok let start from what's in my mind and I was trying to archive.

**Colors looks good but often you can be distracted by them, specially if there is too much mix around.**

Start with "lean configuration", I would like to put some emphasis just on git status when I'm in those kind of director!

With just a few change
- colors (compare as they come by running the configuration) 
- icons (used for the various states)
- add "stashed" scenario that wasn't considered at all
I end up with this configuration 

```
#####################################[ vcs: git status ]######################################
  # Branch icon. Set this parameter to '\uF126 ' for the popular Powerline branch icon.
  typeset -g POWERLEVEL9K_VCS_BRANCH_ICON='\uF126 '
  POWERLEVEL9K_VCS_BRANCH_ICON=${(g::)POWERLEVEL9K_VCS_BRANCH_ICON}

  # Untracked files icon. It's really a question mark, your font isn't broken.
  # Change the value of this parameter to show a different icon.
  #typeset -g POWERLEVEL9K_VCS_UNTRACKED_ICON='?'
  #POWERLEVEL9K_VCS_UNTRACKED_ICON=${(g::)POWERLEVEL9K_VCS_UNTRACKED_ICON}

  # Formatter for Git status.
  #
  # Example output: master ⇣42⇡42 *42 merge ~42 +42 !42 ?42.
  #
  # You can edit the function to customize how Git status looks.
  #
  # VCS_STATUS_* parameters are set by gitstatus plugin. See reference:
  # https://github.com/romkatv/gitstatus/blob/master/gitstatus.plugin.zsh.
  function my_git_formatter() {
    emulate -L zsh

    if [[ -n $P9K_CONTENT ]]; then
      # If P9K_CONTENT is not empty, use it. It's either "loading" or from vcs_info (not from
      # gitstatus plugin). VCS_STATUS_* parameters are not available in this case.
      typeset -g my_git_format=$P9K_CONTENT
      return
    fi

    if (( $1 )); then
      # Styling for up-to-date Git status.
      local       meta='%f'     # default foreground
      local      clean='%10F'   # lightgreen foreground
      local   modified='%11F'   # lightyellow foreground
      local  untracked='%12F'   # lightblue foreground
      local conflicted='%9F'    # lightred foreground
      local    stashes='%13F'   # lightmagenta foreground
    else
      # Styling for incomplete and stale Git status.
      local       meta='%244F'  # grey foreground
      local      clean='%244F'  # grey foreground
      local   modified='%244F'  # grey foreground
      local  untracked='%244F'  # grey foreground
      local conflicted='%244F'  # grey foreground
      local    stashes='%244F'  # grey foreground
    fi

    local res
    local where  # branch or tag
    if [[ -n $VCS_STATUS_LOCAL_BRANCH ]]; then
      res+="${clean}${POWERLEVEL9K_VCS_BRANCH_ICON}"
      where=${(V)VCS_STATUS_LOCAL_BRANCH}
    elif [[ -n $VCS_STATUS_TAG ]]; then
      res+="${meta}#"
      where=${(V)VCS_STATUS_TAG}
    fi

    # If local branch name or tag is at most 32 characters long, show it in full.
    # Otherwise show the first 12 … the last 12.
    (( $#where > 32 )) && where[13,-13]="…"
    res+="${clean}${where//\%/%%}"  # escape %

    # Display the current Git commit if there is no branch or tag.
    # Tip: To always display the current Git commit, remove `[[ -z $where ]] &&` from the next line.
    [[ -z $where ]] && res+="${meta}@${clean}${VCS_STATUS_COMMIT[1,8]}"

    # Show tracking branch name if it differs from local branch.
    if [[ -n ${VCS_STATUS_REMOTE_BRANCH:#$VCS_STATUS_LOCAL_BRANCH} ]]; then
      res+="${meta}:${clean}${(V)VCS_STATUS_REMOTE_BRANCH//\%/%%}"  # escape %
    fi

    # ⇣42 if behind the remote.
    #(( VCS_STATUS_COMMITS_BEHIND )) && res+=" ${clean}⇣${VCS_STATUS_COMMITS_BEHIND}"
    (( VCS_STATUS_COMMITS_BEHIND )) && res+=" ${clean}$(print_icon 'VCS_INCOMING_CHANGES_ICON')${VCS_STATUS_COMMITS_BEHIND}"
    # ⇡42 if ahead of the remote; no leading space if also behind the remote: ⇣42⇡42.
    (( VCS_STATUS_COMMITS_AHEAD && !VCS_STATUS_COMMITS_BEHIND )) && res+=" "
    #(( VCS_STATUS_COMMITS_AHEAD  )) && res+="${clean}⇡${VCS_STATUS_COMMITS_AHEAD}"
    (( VCS_STATUS_COMMITS_AHEAD  )) && res+=" ${clean}$(print_icon 'VCS_OUTGOING_CHANGES_ICON')${VCS_STATUS_COMMITS_AHEAD}"
    # *42 if have stashes.
    #(( VCS_STATUS_STASHES        )) && res+=" ${clean}*${VCS_STATUS_STASHES}"
    (( VCS_STATUS_STASHES        )) && res+=" ${stashes}$(print_icon 'VCS_STASH_ICON')${VCS_STATUS_STASHES}"
    # 'merge' if the repo is in an unusual state.
    #[[ -n $VCS_STATUS_ACTION     ]] && res+=" ${conflicted}${VCS_STATUS_ACTION}"
    # ~42 if have merge conflicts.
    #(( VCS_STATUS_NUM_CONFLICTED )) && res+=" ${conflicted}~${VCS_STATUS_NUM_CONFLICTED}"
    (( VCS_STATUS_NUM_CONFLICTED )) && res+=" ${conflicted}$(echo '\uE727')${VCS_STATUS_NUM_CONFLICTED}"   # '\uF421 ' <- TRIANGLE ||  '\uE727' <- merge 
    # +42 if have staged changes.
    #(( VCS_STATUS_NUM_STAGED     )) && res+=" ${modified}+${VCS_STATUS_NUM_STAGED}"
    (( VCS_STATUS_NUM_STAGED     )) && res+=" ${modified}$(print_icon 'VCS_STAGED_ICON')${VCS_STATUS_NUM_STAGED}"
    # !42 if have unstaged changes.
    #(( VCS_STATUS_NUM_UNSTAGED   )) && res+=" ${modified}!${VCS_STATUS_NUM_UNSTAGED}"
    (( VCS_STATUS_NUM_UNSTAGED   )) && res+=" ${modified}$(print_icon 'VCS_UNSTAGED_ICON')${VCS_STATUS_NUM_UNSTAGED}"
    # ?42 if have untracked files. It's really a question mark, your font isn't broken.
    # See POWERLEVEL9K_VCS_UNTRACKED_ICON above if you want to use a different icon.
    # Remove the next line if you don't want to see untracked files at all.
    #(( VCS_STATUS_NUM_UNTRACKED  )) && res+=" ${untracked}${POWERLEVEL9K_VCS_UNTRACKED_ICON}${VCS_STATUS_NUM_UNTRACKED}"
    (( VCS_STATUS_NUM_UNTRACKED  )) && res+=" ${untracked}$(print_icon 'VCS_UNTRACKED_ICON')${VCS_STATUS_NUM_UNTRACKED}"

    typeset -g my_git_format=$res
  }
  functions -M my_git_formatter 2>/dev/null

  # Disable the default Git status formatting.
  typeset -g POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=false
  # Install our own Git status formatter.
  typeset -g POWERLEVEL9K_VCS_CONTENT_EXPANSION='${$((my_git_formatter(1)))+${my_git_format}}'
  typeset -g POWERLEVEL9K_VCS_LOADING_CONTENT_EXPANSION='${$((my_git_formatter(0)))+${my_git_format}}'
  # Enable counters for staged, unstaged, etc.
  typeset -g POWERLEVEL9K_VCS_{STAGED,UNSTAGED,UNTRACKED,CONFLICTED,COMMITS_AHEAD,COMMITS_BEHIND}_MAX_NUM=-1

  # Icon color.
  typeset -g POWERLEVEL9K_VCS_VISUAL_IDENTIFIER_COLOR=0
  typeset -g POWERLEVEL9K_VCS_LOADING_VISUAL_IDENTIFIER_COLOR=0
  # Custom icon.
  # typeset -g POWERLEVEL9K_VCS_VISUAL_IDENTIFIER_EXPANSION='⭐'
  # Custom prefix.
  # typeset -g POWERLEVEL9K_VCS_PREFIX='%fon '

  # Show status of repositories of these types. You can add svn and/or hg if you are
  # using them. If you do, your prompt may become slow even when your current directory
  # isn't in an svn or hg reposotiry.
  typeset -g POWERLEVEL9K_VCS_BACKENDS=(git)

  # These settings are used for respositories other than Git or when gitstatusd fails and
  # Powerlevel10k has to fall back to using vcs_info.
  typeset -g POWERLEVEL9K_VCS_LOADING_FOREGROUND=8
  typeset -g POWERLEVEL9K_VCS_CLEAN_FOREGROUND=10
  typeset -g POWERLEVEL9K_VCS_MODIFIED_FOREGROUND=11
  typeset -g POWERLEVEL9K_VCS_UNTRACKED_FOREGROUND=12
  typeset -g POWERLEVEL9K_VCS_CONFLICTED_FOREGROUND=9
  typeset -g POWERLEVEL9K_VCS_STASHED_FOREGROUND=13

```  

As I said above, wanted to avoid to much color mixing leads me to discard the **right option** (coming from **POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true**) even if at a first sight it doesn't look so bad.
<img width="1884" alt="branch master   POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true (right) - false (left) " src="https://user-images.githubusercontent.com/8858312/74086401-c0daf380-4a82-11ea-990f-de28237c7152.png">

Than I taught.....
it might be nice mix style and color just for git information the background instead the foreground.
**That would even a better emphasis**
To archive such a goal I just did few changes 
- (literally swap back/fore-ground) 
- still add Stash state for relative git state
- set to basic foreground all the same for every state 
Final configuration as below:

```
#####################################[ vcs: git status ]######################################
  # Branch icon. Set this parameter to '\uF126 ' for the popular Powerline branch icon.
  typeset -g POWERLEVEL9K_VCS_BRANCH_ICON='\uF126 '
  POWERLEVEL9K_VCS_BRANCH_ICON=${(g::)POWERLEVEL9K_VCS_BRANCH_ICON}

  # Untracked files icon. It's really a question mark, your font isn't broken.
  # Change the value of this parameter to show a different icon.
  #typeset -g POWERLEVEL9K_VCS_UNTRACKED_ICON='?'
  #POWERLEVEL9K_VCS_UNTRACKED_ICON=${(g::)POWERLEVEL9K_VCS_UNTRACKED_ICON}

  # Formatter for Git status.
  #
  # Example output: master ⇣42⇡42 *42 merge ~42 +42 !42 ?42.
  #
  # You can edit the function to customize how Git status looks.
  #
  # VCS_STATUS_* parameters are set by gitstatus plugin. See reference:
  # https://github.com/romkatv/gitstatus/blob/master/gitstatus.plugin.zsh.
  function my_git_formatter() {
    emulate -L zsh

    if [[ -n $P9K_CONTENT ]]; then
      # If P9K_CONTENT is not empty, use it. It's either "loading" or from vcs_info (not from
      # gitstatus plugin). VCS_STATUS_* parameters are not available in this case.
      typeset -g my_git_format=$P9K_CONTENT
      return
    fi

    if (( $1 )); then
      # Styling for up-to-date Git status.
      local       meta='%f'    # default foreground
      local      clean='%F'    # default foreground
      local   modified='%F'    # default foreground
      local  untracked='%F'    # default foreground
      local conflicted='%F'    # default foreground
      local    stashes='%F'    # default foreground
    else
      # Styling for incomplete and stale Git status.
      local       meta='%244F'  # grey foreground
      local      clean='%244F'  # grey foreground
      local   modified='%244F'  # grey foreground
      local  untracked='%244F'  # grey foreground
      local conflicted='%244F'  # grey foreground
      local    stashes='%244F'  # grey foreground
    fi

    local res
    local where  # branch or tag
    if [[ -n $VCS_STATUS_LOCAL_BRANCH ]]; then
      res+="${clean}${POWERLEVEL9K_VCS_BRANCH_ICON}"
      where=${(V)VCS_STATUS_LOCAL_BRANCH}
    elif [[ -n $VCS_STATUS_TAG ]]; then
      res+="${meta}#"
      where=${(V)VCS_STATUS_TAG}
    fi

    # If local branch name or tag is at most 32 characters long, show it in full.
    # Otherwise show the first 12 … the last 12.
    (( $#where > 32 )) && where[13,-13]="…"
    res+="${clean}${where//\%/%%}"  # escape %

    # Display the current Git commit if there is no branch or tag.
    # Tip: To always display the current Git commit, remove `[[ -z $where ]] &&` from the next line.
    [[ -z $where ]] && res+="${meta}@${clean}${VCS_STATUS_COMMIT[1,8]}"

    # Show tracking branch name if it differs from local branch.
    if [[ -n ${VCS_STATUS_REMOTE_BRANCH:#$VCS_STATUS_LOCAL_BRANCH} ]]; then
      res+="${meta}:${clean}${(V)VCS_STATUS_REMOTE_BRANCH//\%/%%}"  # escape %
    fi

    # ⇣42 if behind the remote.
    #(( VCS_STATUS_COMMITS_BEHIND )) && res+=" ${clean}⇣${VCS_STATUS_COMMITS_BEHIND}"
    (( VCS_STATUS_COMMITS_BEHIND )) && res+=" ${clean}$(print_icon 'VCS_INCOMING_CHANGES_ICON')${VCS_STATUS_COMMITS_BEHIND}"
    # ⇡42 if ahead of the remote; no leading space if also behind the remote: ⇣42⇡42.
    (( VCS_STATUS_COMMITS_AHEAD && !VCS_STATUS_COMMITS_BEHIND )) && res+=" "
    #(( VCS_STATUS_COMMITS_AHEAD  )) && res+="${clean}⇡${VCS_STATUS_COMMITS_AHEAD}"
    (( VCS_STATUS_COMMITS_AHEAD  )) && res+=" ${clean}$(print_icon 'VCS_OUTGOING_CHANGES_ICON')${VCS_STATUS_COMMITS_AHEAD}"
    # *42 if have stashes.
    #(( VCS_STATUS_STASHES        )) && res+=" ${clean}*${VCS_STATUS_STASHES}"
    (( VCS_STATUS_STASHES        )) && res+=" ${stashes}$(print_icon 'VCS_STASH_ICON')${VCS_STATUS_STASHES}"
    # 'merge' if the repo is in an unusual state.
    #[[ -n $VCS_STATUS_ACTION     ]] && res+=" ${conflicted}${VCS_STATUS_ACTION}"
    # ~42 if have merge conflicts.
    #(( VCS_STATUS_NUM_CONFLICTED )) && res+=" ${conflicted}~${VCS_STATUS_NUM_CONFLICTED}"
    (( VCS_STATUS_NUM_CONFLICTED )) && res+=" ${conflicted}$(echo '\uE727')${VCS_STATUS_NUM_CONFLICTED}"   # '\uF421 ' <- TRIANGLE ||  '\uE727' <- merge 
    # +42 if have staged changes.
    #(( VCS_STATUS_NUM_STAGED     )) && res+=" ${modified}+${VCS_STATUS_NUM_STAGED}"
    (( VCS_STATUS_NUM_STAGED     )) && res+=" ${modified}$(print_icon 'VCS_STAGED_ICON')${VCS_STATUS_NUM_STAGED}"
    # !42 if have unstaged changes.
    #(( VCS_STATUS_NUM_UNSTAGED   )) && res+=" ${modified}!${VCS_STATUS_NUM_UNSTAGED}"
    (( VCS_STATUS_NUM_UNSTAGED   )) && res+=" ${modified}$(print_icon 'VCS_UNSTAGED_ICON')${VCS_STATUS_NUM_UNSTAGED}"
    # ?42 if have untracked files. It's really a question mark, your font isn't broken.
    # See POWERLEVEL9K_VCS_UNTRACKED_ICON above if you want to use a different icon.
    # Remove the next line if you don't want to see untracked files at all.
    #(( VCS_STATUS_NUM_UNTRACKED  )) && res+=" ${untracked}${POWERLEVEL9K_VCS_UNTRACKED_ICON}${VCS_STATUS_NUM_UNTRACKED}"
    (( VCS_STATUS_NUM_UNTRACKED  )) && res+=" ${untracked}$(print_icon 'VCS_UNTRACKED_ICON')${VCS_STATUS_NUM_UNTRACKED}"

    typeset -g my_git_format=$res
  }
  functions -M my_git_formatter 2>/dev/null

  # Disable the default Git status formatting.
  typeset -g POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=false
  # Install our own Git status formatter.
  typeset -g POWERLEVEL9K_VCS_CONTENT_EXPANSION='${$((my_git_formatter(1)))+${my_git_format}}'
  typeset -g POWERLEVEL9K_VCS_LOADING_CONTENT_EXPANSION='${$((my_git_formatter(0)))+${my_git_format}}'
  # Enable counters for staged, unstaged, etc.
  typeset -g POWERLEVEL9K_VCS_{STAGED,UNSTAGED,UNTRACKED,CONFLICTED,COMMITS_AHEAD,COMMITS_BEHIND}_MAX_NUM=-1

  # Icon color.
  typeset -g POWERLEVEL9K_VCS_VISUAL_IDENTIFIER_COLOR=0
  typeset -g POWERLEVEL9K_VCS_LOADING_VISUAL_IDENTIFIER_COLOR=0
  # Custom icon.
  # typeset -g POWERLEVEL9K_VCS_VISUAL_IDENTIFIER_EXPANSION='⭐'
  # Custom prefix.
  # typeset -g POWERLEVEL9K_VCS_PREFIX='%fon '

  # Show status of repositories of these types. You can add svn and/or hg if you are
  # using them. If you do, your prompt may become slow even when your current directory
  # isn't in an svn or hg reposotiry.
  typeset -g POWERLEVEL9K_VCS_BACKENDS=(git)

  # These settings are used for respositories other than Git or when gitstatusd fails and
  # Powerlevel10k has to fall back to using vcs_info.  
  typeset -g POWERLEVEL9K_VCS_LOADING_BACKGROUND=8       # grigio
  typeset -g POWERLEVEL9K_VCS_CLEAN_BACKGROUND=10        # green
  typeset -g POWERLEVEL9K_VCS_MODIFIED_BACKGROUND=11     # yellow
  typeset -g POWERLEVEL9K_VCS_UNTRACKED_BACKGROUND=12    # blue
  typeset -g POWERLEVEL9K_VCS_CONFLICTED_BACKGROUND=9    # red
  typeset -g POWERLEVEL9K_VCS_STASHED_BACKGROUND=13      # magenta

```

<img width="1886" alt="branch vcs_fix   POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true (right) - false (left) " src="https://user-images.githubusercontent.com/8858312/74086399-bfa9c680-4a82-11ea-8e75-d942fb0a89dd.png">

**please be aware that, this final result already include the changes to p10k.zsh file that are in PR
without them the result would be the follow (just add the case with POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true - right side on the previous screenshot)
All file (include .p10.zsh) are now in my brach "vcs_fix" of my fork repository** 


--------------------


Without removing from the if the condition mention in my first post the result would have been
<img width="989" alt="error without vcs_fix" src="https://user-images.githubusercontent.com/8858312/74087746-80ce3d80-4a8f-11ea-820a-de03f6ea0790.png">
It seems like conflict status is not detected by _function _p9k_vcs_render()
**you can find related full file (include .p10k.zsh) in branch "image3" on my fork repository**

**Comment by romkatv** [comment]
I'm confused by your comments. If at all possible, could you please follow the format I suggested in my previous comment? Provide the 3 types of information that I outlined there and nothing else. Use an unmodified powerlevel10k theme when producing the screenshot.

---

It appears that you are defining parameters whose names start with `POWERLEVEL9K_VCS_CONFLICTED_` without defining `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`. These parameters will have no effect. If you want them to have an effect, you must define `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`.

**Comment by jD4rk** [comment]
Ok

I knew it that my first reply would be a mess.....check it now, it might be a little bit more clear.

About your last request (or better the first that I misunderstood)

In my .p10k.zsh file there is no trace of any `POWERLEVEL9K_VCS_CONFLICTED_` apart background and foreground color (**check my provious edited answer**)


In p10k.zsh (plain as it come from configuration)
it's initialized with   `_p9k_declare -b POWERLEVEL9K_VCS_CONFLICTED_STATE 0` and then its value is checked inside function _p9k_vcs_render() in 2 places

-   `if (( _POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING )); then` that I guess it will render the prompt when the relative setting into .p10k.zsh is set to true

-   `if ! _p9k_cache_ephemeral_get "$state" "${(@)cache_key}"; then` that without digging into too much I guess it's fire when in .p10k.zsh file   _typeset -g POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=false_


Honestly I'm a bit confuse as well now because of what you said

> It appears that you are defining parameters whose names start with POWERLEVEL9K_VCS_CONFLICTED_ without defining POWERLEVEL9K_VCS_CONFLICTED_STATE=true. These parameters will have no effect. If you want them to have an effect, you must define POWERLEVEL9K_VCS_CONFLICTED_STATE=true.
Why there is a condition about it inside if statement where we are evaluate if the current state is conflicted or not?!

Taking as example one portion of code
```
  if (( _POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING )); then
    if [[ -z $state ]]; then
      if [[ $VCS_STATUS_HAS_CONFLICTED == 1 && $_POWERLEVEL9K_VCS_CONFLICTED_STATE == 1 ]]; then
        state=CONFLICTED
      elif [[ $VCS_STATUS_HAS_STAGED != 0 || $VCS_STATUS_HAS_UNSTAGED != 0 ]]; then
        state=MODIFIED
      elif [[ $VCS_STATUS_HAS_UNTRACKED != 0 ]]; then
        state=UNTRACKED
      else
        state=CLEAN
      fi
    fi
```

Why mixing check about  VCS_STATUS_*states (that comes from gitstatus) with _POWERLEVEL9K_VCS_* that are define by our own.
If, the purpose would have to suppress highlight about conflict state in the prompt why don't do the same for staged, untracked, modified...etc etc

kind of
```
  if (( _POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING )); then
    if [[ -z $state ]]; then
      if [[ $VCS_STATUS_HAS_CONFLICTED == 1 && $_POWERLEVEL9K_VCS_CONFLICTED_STATE == 1 ]]; then
        state=CONFLICTED
      elif [[ $VCS_STATUS_HAS_STAGED != 0 || $VCS_STATUS_HAS_UNSTAGED != 0  && _POWERLEVEL9K_VCS_STAGED_STATE == 1 ]]; then
        state=MODIFIED
      elif [[ $VCS_STATUS_HAS_UNTRACKED != 0 && _POWERLEVEL9K_VCS_UNTRACKED_STATE == 1]]; then
        state=UNTRACKED
      else
        state=CLEAN
      fi
    fi
```
declaring of course all of them with 0 value as default and add into your .p10k.zsh 
  typeset -g POWERLEVEL9K_VCS_CONFLICTED_STATE=true
  typeset -g POWERLEVEL9K_VCS_STAGED_STATE=true
  typeset -g POWERLEVEL9K_VCS_UNTRACKED_STATE=true
and so on, so on...


I guess make it only for the conflict state brings to kind of semantic error and consequence mis-understanding.

**Comment by romkatv** [comment]
> In my .p10k.zsh file there is no trace of any `POWERLEVEL9K_VCS_CONFLICTED_` apart background and foreground color (**check my provious edited answer**)

Let me quote my previous comment and annotate it.

> It appears that you are defining parameters whose names start with `POWERLEVEL9K_VCS_CONFLICTED_`

This says that your config defines at least one parameter whose name starts with `POWERLEVEL9K_VCS_CONFLICTED_`. **Is this true?**

> without defining `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`

This says that your config does not define `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`. **Is this true?**

> These parameters will have no effect.

I can assure you that this statement is true if the two previous statements are true.

> If you want them to have an effect, you must define `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`.

Likewise.

---

The title of the issue says there is "possible error in vcs". Could you clarify what it is? Please use the following structure when describing this possible error. PLEASE DO EXACTLY THIS OR I WILL NOT UNDERSTAND YOU (sorry about the caps but it's really important).

1. Use unmodified version of powerlevel10k.
2. Take a screenshot of your prompt.
3. Explain what on the screenshot signifies a possible error in vcs.
4. Post your complete `~/.p10k.zsh`.

Do NOT do the following:

- Quote any source code from powerlevel10k.
- Modify any source code from powerlevel10k.
- Suggest how the error can be fixed in powerlevel10k.

These points aren't bad per se but it's better to postpone discussing them until I understand what the error is.

**Comment by jD4rk** [comment]
Ok, 

after what you said about the function of `POWERLEVEL9K_VCS_CONFLICTED_STATE=true`
obviously now things are more clear!

In a plain config that comes by run the command "p10k configure" no matters what setting are you changing (regard back/fore-ground color, icons and whatever) missing the above declarations leads you to do not have a merge conflict properly displayed.



Here is output from and untouched config file (and also no changes of powerlevel10k source code)
<img width="1120" alt="Screenshot 2020-02-08 20 10 53" src="https://user-images.githubusercontent.com/8858312/74090750-52f8f100-4aaf-11ea-9abe-abac479303d0.png">
I'm aspecting, for _merge conflict state_,  the same behavior as per _changed file state_ (that color the whole segment in yellow).
Instead, how shows in the above image, when I'm in a merge conflict state, only a "red error indicator" is appended to the segment which is still colored by the previous state.

Of course, knowing now the scope of `POWERLEVEL9K_VCS_CONFLICTED_STATE=true` by just add into config file the follow
```
  # Enable hightlight for merge conflict git directory
  typeset -g POWERLEVEL9K_VCS_CONFLICTED_STATE=true
```
turn the prompt as I'm aspected to see...
but something looks anyway a little bit weird (**red on red**).
Here is what I'm talking about
<img width="616" alt="Screenshot 2020-02-08 20 23 57" src="https://user-images.githubusercontent.com/8858312/74090873-f0a0f000-4ab0-11ea-9e5f-98e988d3ab2c.png">
even by try to force specific foreground color   `typeset -g POWERLEVEL9K_VCS_CONFLICTED_FOREGROUND=0` won't fix the issue.


Now,  what I'm aspecting to see is something like:
<img width="446" alt="Screenshot 2020-02-08 20 29 36" src="https://user-images.githubusercontent.com/8858312/74090923-cb60b180-4ab1-11ea-9dad-a54eec381897.png">
which I've been able to archived by follow settings in the conf file
```
   local conflicted='%0F' # red foreground

  # Disable the default Git status formatting.
  typeset -g POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true

  # Enable highlight for merge conflict git directory
  typeset -g POWERLEVEL9K_VCS_CONFLICTED_STATE=true
```

Therefore those change, for what I've understand, changes also the source where the data come from,
(it can be clearly understood by check that the icons, now come from the config file instead from the icons file)

In conclusion,
I don't know if all of this is just me, be too picky when I'm trying to do something 
or if it has been made by purpose (I don't believe so neither).
Moreover, set POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true/false shouldn't effect the way as the segment is showed but only how/where the data has grabbed from (as the state of the directory is still the same in both the cases).

Hope now, I explain my mind and my concern clearly

**Comment by romkatv** [comment]
> Here is output from and untouched config file (and also no changes of powerlevel10k source code)
> <img alt="Screenshot 2020-02-08 20 10 53" width="1120" src="https://user-images.githubusercontent.com/8858312/74090750-52f8f100-4aaf-11ea-9abe-abac479303d0.png">

This ☝️ looks as expected.

> ```
>   # Enable hightlight for merge conflict git directory
>   typeset -g POWERLEVEL9K_VCS_CONFLICTED_STATE=true
> ```
> <img alt="Screenshot 2020-02-08 20 23 57" width="616" src="https://user-images.githubusercontent.com/8858312/74090873-f0a0f000-4ab0-11ea-9e5f-98e988d3ab2c.png">

This ☝️ looks as expected.

> even by try to force specific foreground color `typeset -g POWERLEVEL9K_VCS_CONFLICTED_FOREGROUND=0` won't fix the issue.

The parameter you are looking for is called `POWERLEVEL9K_VCS_ACTIONFORMAT_FOREGROUND`.

> Therefore those change, for what I've understand, [`POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING`] changes also the source where the data come from

It does not. `POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING` makes the content of `vcs` empty when data comes from gitstatus. This parameter exists to avoid wasting time on assembling the content of `vcs` twice.

> Moreover, set POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING=true/false shouldn't effect the way as the segment is showed but only how/where the data has grabbed from (as the state of the directory is still the same in both the cases).

It's the opposite. `POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING` does what it's name suggests. It disables the builtin formatter for vcs segment when data is pulled from gitstatus. It doesn't do anything else.

I don't recommend changing the value of `POWERLEVEL9K_VCS_DISABLE_GITSTATUS_FORMATTING` in `~/.p10k.zsh`. It's **much easier** to customize the format of `vcs` by editing the code of `my_git_formatter`. If you want the background to turn red when there are conflicting changes, simply add `%K{red}` when `VCS_STATUS_NUM_CONFLICTED` is not zero. When using `my_git_formatter`, you no longer need to hunt down obscure `POWERLEVEL9K` parameters and guess what they mean.

It doesn't seem like there is anything for me to do here, so I'm going to close the issue.
